share/patch_postgresql_db_zonemaster_backend_ver_7.0.0.pl

```diff
@@ -0,0 +1,27 @@
+use strict;
+use warnings;
+use utf8;
+use Data::Dumper;
+use Encode;
+
+use DBI qw(:utils);
+
+use Zonemaster::Backend::Config;
+use Zonemaster::Backend::DB::PostgreSQL;
+
+my $config = Zonemaster::Backend::Config->load_config();
+if ( $config->DB_engine ne 'PostgreSQL' ) {
+    die "The configuration file does not contain the PostgreSQL backend";
+}
+my $dbh = Zonemaster::Backend::DB::PostgreSQL->from_config( $config )->dbh;
+
+sub patch_db {
+
+    ####################################################################
+    # TEST RESULTS
+    ####################################################################
+    $dbh->do( 'ALTER TABLE test_results ADD COLUMN undelegated integer NOT NULL DEFAULT 0' );
```

Comment:
Why is the `undelegated` column of type integer here and of type boolean in the creation script?

Reply:
Also I see we are marking all previous tests as delegated even if they were not, is that okay?

Reply:
It should be integer in both cases. I missed to change in the creation script. Corrected now.

Good observation that all previous tests will classified as delegated. For previous tests in MySQL or SQLite databases, all tests are stored as undelegated even when they are not.

One option is to let it be as is since this classification has not been used. Another is to create a conversion script for all three database engines to rewrite the tests to make the classification to be correct.

What do you think?

Reply:
I think that it would be nice to have the correct data in database, but I understand that it requires some work to make the update scripts.

You made me realize that the GUI does not allow to filter for undelegated / delegated tests (or else I missed it), is that something we could want? (And could make the update scripts worth writing). Should I open an issue on the GUI regarding that feature?

Reply:
> I think that it would be nice to have the correct data in database, but I understand that it requires some work to make the update scripts.

I agree to both points. I am not sure that it is worth the effort, but we could think about it before the release. When this PR is merged, I will create an issue that lifts this up and we can decide later.

> You made me realize that the GUI does not allow to filter for undelegated / delegated tests (or else I missed it), is that something we could want? (And could make the update scripts worth writing). Should I open an issue on the GUI regarding that feature?

That was my plan to do after this is merged, but go ahead! My idea is as follows:
* In the normal case, the undelegated tests are unwanted because the are really not comparable with any other test unless the domain name and DS settings are equal.
* By default only delegated tests are to be shown, with a buttons saying "all historic tests" and "only undelegated tests".
* I was thinking if the result JSON of method "get_test_history" should contain the delegated value to make it possible for the GUI to swap between the selections.

Reply:
> When this PR is merged, I will create an issue that lifts this up and we can decide later.

Okay this is fine.

> I was thinking if the result JSON of method "get_test_history" should contain the delegated value to make it possible for the GUI to swap between the selections.

I think that could simplify the GUI implementation indeed.

I created this issue from your comments https://github.com/zonemaster/zonemaster-gui/issues/251 so the discussion can move there if needed.